src/lustre/lustreDesugarChooseOps.ml

```diff
@@ -180,6 +180,29 @@
    | CallParam (pos, id, types, expr_list) ->
      let expr_list, gen_nodes = List.map (desugar_expr ctx node_name) expr_list |> List.split in
      CallParam (pos, id, types, expr_list), List.flatten gen_nodes
+
+ let desugar_contract_item ctx node_name ci = 
+    match ci with 
+      | A.GhostVars (pos, lhs, e) -> 
+        let e, gen_nodes = desugar_expr ctx node_name e in 
+        A.GhostVars (pos, lhs, e), gen_nodes
+      | Assume (pos, name, b, e) ->
+        let e, gen_nodes = desugar_expr ctx node_name e in 
+        Assume (pos, name, b, e), gen_nodes
+      | Guarantee (pos, name, b, e) -> 
+        let e, gen_nodes = desugar_expr ctx node_name e in 
+        Guarantee (pos, name, b, e), gen_nodes
+      | GhostConst _ 
+      | Mode _ 
+      | ContractCall _ 
```

Comment:
Choose op desugaring should also happen in the `require` and `ensure` clauses of a mode, and in the arguments of contract calls.

Reply:
Addressed. Also added desugaring for `ContractNodeDecl`s